uncover/session.go

```diff
@@ -37,9 +40,33 @@ func NewSession(keys *Keys, retryMax, timeout int) (*Session, error) {
 	client := retryablehttp.NewWithHTTPClient(httpclient, options)
 
 	session := &Session{
-		Client:   client,
-		Keys:     keys,
-		RetryMax: retryMax,
+		Client:     client,
+		Keys:       keys,
+		RetryMax:   retryMax,
+		RateLimits: &ratelimit.MultiLimiter{},
+	}
+
+	var err error
+	rateLimitOpts := &ratelimit.Options{
+		MaxCount:    uint(retryMax),
+		Duration:    time.Duration(delay),
+		IsUnlimited: delay == 0,
+	}
+
+	rateLimitOpts.Key = engines[0]
```

Comment:
@xm1k3 I think Maxcount should be 1 here and retryMax is configured should be configure in`retryablehttp.Client`